### PR TITLE
feat: add a simple cypress setup

### PR DESCRIPTION
**Pull request by Andy-Rennie**
Add a simple cypress test that loads the package onto a page

### CODE REVIEW

File: cypress/support/built-script.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import * as PurpleDot from '../../src/index';
```
Comment (domkck) [inline cypress/support/built-script.ts:1]: To make this more e2e, is there a way to actually install the package as a dependency and use it like an integration would?

File: package.json [hunk 2]
```diff
@@ -9,7 +9,9 @@
     "lint": "rome ci src/",
     "format": "rome check --apply-unsafe src/ && rome format --write src/",
     "test": "vitest",
-    "coverage": "vitest run --coverage"
+    "coverage": "vitest run --coverage",
+    "cy:open": "cd cypress/support && npx http-server . -p 8080& cd ../.. && cypress open",
+    "cy:run": "cd cypress/support && npx http-server . -p 8080& cd ../.. && cypress run"
```
Comment (domkck) [inline package.json:14]: Does this need to build the `build-script.ts` script?

File: cypress/e2e/spec.cy.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+describe('template spec', () => {
+  it('passes', () => {
+    cy.visit('http://127.0.0.1:8080/dist/index')
```
Comment (domkck) [inline cypress/e2e/spec.cy.js:3]: Does there need to be a `index.html` file checked in?

File: cypress/support/built-script.ts (hunk 1 above)
Comment (public) [reply cypress/support/built-script.ts:1]: Cypress has a feature called "[Component Testing](https://docs.cypress.io/guides/component-testing/overview)" which might be more appropriate for this use case.

File: package.json (hunk 2 above)
Comment (public) [inline package.json:14]: Needs adding to the circle config too?

File: cypress/support/built-script.ts (hunk 1 above)
Comment (Andy-Rennie) [reply cypress/support/built-script.ts:1]: For installing the actual package:
- I did start with importing, but you need to get the code into the window as opposed to just the cypress window surrounding the window
- To install the package we would need the package to see it as another package, which I might be able to do? The build step is the main thing though and this gets pretty close

- I did try the component testing but it looked like you needed to pick React or Vue and then it gives you something to mount into to test individual React or Vue components so didn't seem like a good fit

File: cypress/e2e/spec.cy.js (hunk 3 above)
Comment (Andy-Rennie) [reply cypress/e2e/spec.cy.js:3]: No webpack generates that for us

File: package.json [hunk 4]
```diff
@@ -21,14 +23,33 @@
...
     "jsdom": "22.1.0",
+    "lodash": "^4.17.21",
+    "minimist": "^1.2.8",
     "rome": "12.1.3-nightly.f65b0d9",
+    "ts-loader": "^9.4.4",
+    "tslib": "^2.6.0",
     "typescript": "^5.1.6",
     "vitest": "^0.33.0",
+    "webpack": "^5.88.2",
+    "webpack-cli": "^5.1.4",
```
Comment (domkck) [inline package.json:52]: What are all these dev dependencies? Why do we need them?

**Pull request merged**